test/unit/ngsiv2/bidirectionalityHttp-test.js

```diff
@@ -0,0 +1,258 @@
+                }
+            };
+
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/subscriptions', function(body) {
+
+                    var expectedBody = utils.readExampleFile(
+                    './test/unit/ngsiv2/subscriptionRequests/bidirectionalSubscriptionRequest.json');
+                    // Note that expired field is not included in the json used by this mock as it is a dynamic
+                    // field. The following code performs such calculation and adds the field to the subscription
+                    // payload of the mock.
+                    if (!body.expires)
+                    {
+                        return false;
+                    }
+                    else if (moment(body.expires, 'YYYY-MM-DDTHH:mm:ss.SSSZ').isValid())
+                    {
+                        expectedBody.expires = moment().add(config.deviceRegistrationDuration);
+                        var expiresDiff = moment(expectedBody.expires).diff(body.expires, 'milliseconds');
+                        if (expiresDiff < 500) {
+                            delete expectedBody.expires;
+                            delete body.expires;
+
+                            return JSON.stringify(body) === JSON.stringify(expectedBody);
+                        }
+
+                        return false;
+                    }
+                    else {
+                        return false;
+                    }
+                })
+                .reply(201, null, {'Location': '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext', utils.readExampleFile(
+                    './test/contextRequests/createBidirectionalDevice.json'))
+                .reply(200, '{}');
+
+            iotagentUl.start(config, function(error) {
+                request(provisionOptions, function(error, response, body) {
+                    done();
+                });
+            });
+        });
+
+        it('should return a 200 OK', function(done) {
+            request(notificationOptions, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(200);
+                done();
+            });
+        });
+
+        it('should leave the data in the polling queue', function(done) {
+            request(notificationOptions, function(error, response, body) {
+                iotAgentLib.commandQueue('smartGondor', '/gardens', 'MQTT_2', function(error, list) {
+                    should.not.exist(error);
+
+                    list.commands.length.should.equal(3);
```

Comment:
Of the 3 items, I understand that latitude is one, longitude is other... but, what about other one? What is?